services/frontend-service/src/ui/components/FormattedDate/FormattedDate.tsx

```diff
@@ -0,0 +1,80 @@
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright 2023 freiheit.com*/
+import React, { useEffect } from 'react';
+
+const getRelativeDate = (current: Date, target: Date | undefined): string => {
+    if (!target) return '';
+    const elapsedTime = current.valueOf() - target.valueOf();
+    const msPerMinute = 1000 * 60;
+    const msPerHour = msPerMinute * 60;
+    const msPerDay = msPerHour * 24;
+    const msPerMonth = msPerDay * 30;
+    const msPerYear = msPerDay * 365;
+
+    if (elapsedTime < msPerMinute) {
+        return 'just now';
+    } else if (elapsedTime < msPerHour) {
+        return Math.round(elapsedTime / msPerMinute) === 1
+            ? `1 minute ago`
+            : `${Math.round(elapsedTime / msPerMinute)} minutes ago`;
+    } else if (elapsedTime < msPerDay) {
+        return Math.round(elapsedTime / msPerHour) === 1
+            ? '1 hour ago'
+            : `${Math.round(elapsedTime / msPerHour)} hours ago`;
+    } else if (elapsedTime < msPerMonth) {
+        return Math.round(elapsedTime / msPerDay) === 1
+            ? '~ 1 day ago'
+            : `~ ${Math.round(elapsedTime / msPerDay)} days ago`;
+    } else if (elapsedTime < msPerYear) {
+        return Math.round(elapsedTime / msPerMonth) === 1
+            ? '~ 1 month ago'
+            : `~ ${Math.round(elapsedTime / msPerMonth)} months ago`;
+    } else {
+        return Math.round(elapsedTime / msPerYear) === 1
+            ? '~ 1 year ago'
+            : `~ ${Math.round(elapsedTime / msPerYear)} years ago`;
+    }
+};
+
+export const useRelativeDate = (createdAt: Date | undefined): string => {
+    const [relativeDate, setRelativeDate] = React.useState(getRelativeDate(new Date(), createdAt));
+    useEffect(() => {
+        const handle = setInterval(() => setRelativeDate(getRelativeDate(new Date(), createdAt)), 20000);
+        return () => clearInterval(handle);
+    }, [createdAt]);
+    return relativeDate;
+};
+
+export const FormattedDate: React.FC<{ createdAt: Date; className?: string }> = ({ createdAt, className }) => {
+    // Adds leading zero to get two digit day and month
+    const twoDigit = (num: number): string => (num < 10 ? '0' : '') + num;
+    // date format (ISO): yyyy-mm-dd, with no leading zeros, month is 0-indexed.
+    // createdAt.toISOString() can't be used because it ignores the current time zone.
+    const formattedDate = `${createdAt.getFullYear()}-${twoDigit(createdAt.getMonth() + 1)}-${twoDigit(
+        createdAt.getDate()
+    )}`;
+    const relativeDate = useRelativeDate(createdAt);
+
+    // getHours automatically gets the hours in the correct timezone. in 24h format (no timezone calculation needed)
+    const formattedTime = `${createdAt.getHours()}:${createdAt.getMinutes()}`;
+
+    return (
+        <span className={className} title={createdAt.toString()}>
+            {formattedDate + ' @ ' + formattedTime + ' | '}
```

Comment:
I don't think we need to display the full date and the relative date and have the full date again on hover. I would suggest to display only the relative time, without the seperate <i> tag and have the full time on hover.

Reply:
done :+1: